jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -316,18 +316,21 @@ async def job_submissions_agent_pending(
     logger.trace(f"query = {render_sql(query)}")
     rows = await database.fetch_all(query)
 
-    try:
-        response = [
-            PendingJobSubmission(
-                **row,
-                job_script_data_as_string=s3man_jobscripts[str(row.get("job_script_id"))],
-            )
-            for row in rows
-        ]
-    except KeyError:
-        list_ids = (str(row.get("job_script_id")) for row in rows)
-        missing_ids = {id for id in list_ids if id not in s3man_jobscripts}
-        message = f"JobScript file(s) not found, the missing ids are: {', '.join(missing_ids)}"
+    response = []
+    missing_ids = set()
+
+    for row in rows:
```

Comment:
Something to keep in mind and to address in the future, is that we will need to batch the pending files. If we end up having *many* pending files, trying to pull all of them down at once will be very slow and error prone.

Instead, we should think about changing this endpoint to have a maximum number that it pulls down at a time.